frontend/src/pages/modelServing/screens/projects/nimUtils.ts

```diff
@@ -123,3 +124,59 @@ export const updateServingRuntimeTemplate = (
   }
   return updatedServingRuntime;
 };
+
+export const getNIMResourcesToDelete = async (
+  projectName: string,
+  servingRuntime: ServingRuntimeKind,
+): Promise<Promise<void>[]> => {
+  const resourcesToDelete: Promise<void>[] = [];
+  try {
+    const count = await fetchInferenceServiceCount(projectName);
+    if (count === 0) {
+      return resourcesToDelete;
+    }
+    const pvcName = servingRuntime.spec.volumes?.find((vol) =>
+      vol.persistentVolumeClaim?.claimName.startsWith('nim-pvc'),
+    )?.persistentVolumeClaim?.claimName;
+
+    let nimSecretName: string | undefined;
+    let imagePullSecretName: string | undefined;
+
+    servingRuntime.spec.containers.forEach((container) => {
+      container.env?.forEach((env) => {
+        const secretName = env.valueFrom?.secretKeyRef?.name;
+        if (secretName === 'nim-secret') {
+          nimSecretName = secretName;
+        } else if (secretName === 'nim-ngc-secret') {
+          imagePullSecretName = secretName;
+        }
+      });
+    });
+
+    if (pvcName) {
+      resourcesToDelete.push(deletePvc(pvcName, projectName).then(() => undefined));
+    }
+
+    if (nimSecretName && imagePullSecretName && count === 1) {
+      resourcesToDelete.push(
+        deleteSecret(projectName, nimSecretName).then(() => undefined),
+        deleteSecret(projectName, imagePullSecretName).then(() => undefined),
+      );
+    }
```

Comment:
I don't think this is actually working 😞  -- I still have my secrets left over.

Reply:
As discussed -- this is the highest priority issue -- the other two can be follow up PRs (if needed)... this blocks merging this PR.

Reply:
Fixed